### ISSUE TITLE
Authcfg support for PostGIS Rasters

### ISSUE THREAD

**Issue by borysiasty** [qgis/QGIS#31045]
PostGIS Rasters can be only loaded using connections with plain username & password, not the safe authentication configurations. This is because the GDAL provider doesn't use the authentication system to resolve the authcfg uri parameter, but it badly limits the use of PostGIS Raster to either not stored or unencrypted passwords (what is deprecated since quite a while in my impression).

**Comment by elpaso** [comment]
Wierd: I tested it with authcfg and it worked. I'll test it again.

**Comment by elpaso** [comment]
for reference: https://github.com/qgis/QGIS/blob/master/src/core/providers/gdal/qgsgdalprovider.cpp#L271

**Comment by elpaso** [comment]
Authcfg is  supported (to some extent) by GDAL provider, if it does not work with PG rasters it's a bug.

**Comment by borysiasty** [comment]
@elpaso ah shame on me! Honestly I've never found GDAL trying to resolve it, so I din't even look to the code. I'll look at it!

**Comment by elpaso** [comment]
@borysiasty just to be clear: you did actually test a postgis GDAL raster layer with authcfg and it didn't work? 
Because I re-tested it and it works correctly here.

**Comment by borysiasty** [comment]
@elpaso Exactly. And now I see the reason is the dataSourceUri format... 

In the QgsGdalProvider::dataSourceUri method there is a regex expecting the authcfg value to be delimited with apostrophes:   

https://github.com/qgis/QGIS/blob/master/src/core/providers/gdal/qgsgdalprovider.cpp#L275

But when I click on an item in the browser, I got the URI without that apostrophes:   

Invalid Layer : GDAL provider Cannot open GDAL dataset PG: dbname='qgis' host=localhost port=5432 sslmode=allow authcfg=h713398 mode=2 schema='public' column='rast' table='dem':

If I temporarily change the regex, everything works. 

Tested on master/Debian/GDAL 2.4.0, but the same problem was alywas with DBManager in earlier versions on Debian and Windows.

I wonder how it's possible you have another URI format?
